src/iso9660.rs

```diff
@@ -49,6 +49,19 @@ pub struct IsoFs {
 impl IsoFs {
     pub fn from_file(mut file: fs::File) -> Result<Self> {
         let descriptors = get_volume_descriptors(&mut file)?;
+        match &descriptors[0] {
+            VolumeDescriptor::Boot(_boot_volume_descriptor) => {}
+            VolumeDescriptor::Primary(primary_volume_descriptor) => {
+                if i64::from(primary_volume_descriptor.volume_space_size)
+                    * i64::from(primary_volume_descriptor.logical_block_size)
+                    > file.metadata()?.len().try_into().unwrap()
+                {
+                    bail!("ISO Image is wrong and can't be modified");
```

Comment:
"wrong" isn't very specific; let's say what the problem actually is.  And we don't know that the caller's intent is to modify the ISO; they might just be reading it.